### DISCUSSION TITLE
Emacs@30 Seems to not recognise function key on OSX

### DISCUSSION THREAD

**Discussion by Johnny-Nicola** [d12frosted/homebrew-emacs-plus#583] [Q&A]
Hi...

Have just moved over to emacs+@30 but having real issue recognising my Mac function key that I normally map to `Hyper` key in Spacemacs. All works well in my previous build: `GNU Emacs 28.2 (build 1, x86_64-apple-darwin22.3.0, Carbon Version 169 AppKit 2299.4) of 2023-04-20
`
```
osx :variables
      osx-command-as       'super
      osx-option-as        'meta
      osx-control-as       'control
      osx-function-as      'hyper
```
Only the function key is not recognised... other keys are good.
Have I installed incorrectly?

**Comment by d12frosted** [comment]
Hey. I am not able to test that by myself now. But a small question to clarify. What is the "previous" build exactly? Is it Emacs+ or something else? If it's Emacs+, then I advice to check changelog/mailing lists as Emacs+ does nothing extra for key handling - you get regular GNU Emacs.

**Reply by Johnny-Nicola** [reply]
No... the previous working version is emacs-mac (emacs-28.2-mac-9.1) I've reverted back to that for now. You want anything logging out? :)

**Reply by d12frosted** [reply]
That could be the root cause. You see, `emacs-mac` is GNU Emacs with some major modifications to get better integration with macos. Emacs+ is basically GNU Emacs without those modifications.

But then I look into your variables and I am not sure where they come from. In GNU Emacs we have `mac-*-modifier` variables. And when I set `mac-function-modifier` to `'meta`, it acts as meta key :) Could it be that `osx-*-as` comes from `emacs-mac` or some 3rd party package?
